docs/gen_api.py

```
@@ -0,0 +1,174 @@
+            if depth == 0:
+                end = i
+                break
+        elif start is not None:
+            depth += line.count("[") - line.count("]")
+            if depth <= 0:
+                end = i
+                break
+    if start is None:
+        raise ValueError("nav key not found in zensical.toml")
+    new_block = "nav = " + nav_to_toml_str(new_nav) + "\n"
+    return "".join(lines[:start] + [new_block] + lines[end + 1:])
```

Ruff (0.15.11)

[warning] Consider `[*lines[:start], new_block, *lines[end + 1:]]` instead of concatenation

Replace with `[*lines[:start], new_block, *lines[end + 1:]]`

(RUF005)